Updated Documentation for Release v2024.6.0 (#208)
hoppscotch/docs: documentation/clients/cli/overview.mdx

- The hopp test command recursively goes through each request in the collection and runs them, validating the responses with the test script provided in each request. Hence, the order of execution is the same as the order specified in the collection structure.
- If upon executing the command, a failed assertion (a failing test case) has occurred, the command will give a non-zero exit code and 0 exit code if all tests have passed.
- Unless there was a network error (for example, DNS resolution errors or network Connectivity Issues), the test script will be running and it is up to the test script to define what happens to error status codes. Non-200 status codes are still considered valid responses for test script execution.

```bash
hopp test [-e <environment file>] [-d <delay_in_ms> ] <hoppscotch collection file>
```

### Running Collections and Environments on Cloud

> You can now connect your Hoppscotch Cloud account with the CLI using Persoanl Access Token and execute collections and environments directly by specifying their respective **IDs** in the command. It's important to note that Hoppscotch Cloud functions as the default server in this scenario.

Reviewer: ```suggestion
You can now connect your Hoppscotch Cloud account with the CLI using Personal Access Token and execute collections and environments directly by specifying their respective **IDs** in the command. It's important to note that Hoppscotch Cloud functions as the default server in this scenario.
```

```bash
hopp test  [-e <environment id>] [-d <delay_i_ms>] <hoppscotch collection id> [--token <access_token>]
```

### Running Collections and Environments on Self Hosted instance

To establish a connection with self-hosted Hoppscotch instances, you'll need to specify the server URL along with the access token and IDs to run collections and environments on CLI.

```bash
hopp test  [-e <environment id>] [-d <delay_i_ms>] <hoppscotch collection id> [--token <access_token>] [--server <server url>]
```

### Arguments

- `hoppscotch collection id` : Each collection created in a Hoppscotch workspace is given a unique identifier known as the Collection ID. Collection IDs for each collection can be found under **“Details”** tab inside Collection **“Properties”**.

<Tip> You can directly copy the command with the auto-populated **Collection ID** by navigating to the **“CLI"** tab within the **“Run Collection”** action found in the context menu. </Tip>
    
- `environment id` : Similar to Collection IDs, each environment created in a Hoppscotch workspace is assigned a unique identifier known as the Environment ID.

> - `delay_i_ms` : Represnts a time interval (in milliseconds) to pause execution of API requests before within a collection.
> - `access token` : It is a secure, unique identifier used to authenticate a user's access to their Hoppscotch account and its resources like collections, environements, requests and other entities. [Learn more about retrieving your PAT by checking our documentation](/documentation/features/pat)

Reviewer: ```suggestion
- `delay_i_ms` : Represents a time interval (in milliseconds) to pause execution of API requests before within a collection.
- `access token` : It is a secure, unique identifier used to authenticate a user's access to their Hoppscotch account and its resources like collections, environments, requests and other entities. [Learn more about retrieving your PAT by checking our documentation](/documentation/features/pat)
```

## Options

| Option          | Description                                                                                                       |
| ------          | ------------------------------------------------------------------------------------------------------------------|                    

> | `-h`            | Gives a list of associated commands and their descriptions                                                        |
> | `-v`            | Displays the curremt version of the CLI                                                                           |
> |`--env or -e`    | Accepts environments in all the formats present in [Environment section](/documentation/clients/cli#environment). |
> |`--delay` or `-d`| Used to defer the execution of requests in a collection.                                                          |
> |`--token`        | Expects a PAT value to be passed for establishing connection with Hoppscotch account.                             |
> |`--server`       | Demands a server URL where self-hosted instance is running.                                                       |

Reviewer: ```suggestion
| `-h`            | Gives a list of associated commands and their descriptions                                                        |
| `-v`            | Displays the current version of the CLI                                                                           |
|`--env or -e`    | Accepts environments in all the formats present in [Environment section](/documentation/clients/cli#environment). |
|`--delay` or `-d`| Used to defer the execution of requests in a collection.                                                          |
|`--token`        | Expects a PAT value to be passed for establishing a connection with the Hoppscotch account.                             |
|`--server`       | Demands a server URL where the self-hosted instance is running.                                                       |
```
